### PR TITLE
HDDS-10719. Empty ETag for key created outside of S3

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Avoid introducing empty `ETag` in response for keys created outside of S3 Gateway (e.g. via Ozone CLI).

https://issues.apache.org/jira/browse/HDDS-10719

## How was this patch tested?

Added assertion in existing Robot test.

CI:
https://github.com/adoroszlai/ozone/actions/runs/8758764190

### CODE REVIEW

File: hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/BasicOmKeyInfo.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ private BasicOmKeyInfo(Builder b) {
     this.modificationTime = b.modificationTime;
     this.replicationConfig = b.replicationConfig;
     this.isFile = b.isFile;
-    this.eTag = b.eTag;
+    this.eTag = b.eTag != null && !b.eTag.isEmpty() ? b.eTag : null;
```
[the remark this answers is not in the document]
Comment (ivandika3) [reply hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/BasicOmKeyInfo.java:54]: Or we might instead use an protobuf field existence check for eTag field on `BasicOmKeyInfo#getFromProtobuf`.

```java
if (basicKeyInfo.hasETag()) {
  builder.setETag(basicKeyInfo.getETag());
}
```

**Review by ivandika3 — COMMENTED** [review]
Thank you for the fix. I left a comment.

**Review by ivandika3 — APPROVED** [review]
Thank you for the update. LGTM +1.

**Comment by adoroszlai** [comment]
Thanks @ivandika3 for the review.

**Pull request merged**